corehq/apps/aggregate_ucrs/column_specs.py

```diff
@@ -280,15 +281,43 @@ class SingleFieldColumnProperties(jsonobject.JsonObject):
     referenced_column = jsonobject.StringProperty(required=True)
 
 
-class SumColumnAdapter(SecondaryColumnAdapter):
+class SimpleAggregationAdapater(six.with_metaclass(ABCMeta, SecondaryColumnAdapter)):
     """
-    A SecondaryColumnAdapter class that sums the values of a given column in the
-    secondary table.
+    Generic SecondaryColumnAdapter class that does a passed-in sqlalchemy aggregation.
     """
     config_spec = SingleFieldColumnProperties
 
+    @abstractproperty
+    def sqlalchemy_fn(self):
+        pass
+
+    def to_sqlalchemy_query_column(self, sqlalchemy_table, aggregation_params):
+        return self.sqlalchemy_fn(sqlalchemy_table.c[self.properties.referenced_column])
```

Comment:
You could make this support joins without aggregation if you did something like this:
```
ref_col = sqlalchemy_table.c[self.properties.referenced_column]
return self.sqlalchemy_fn(ref_col) if self.sqlalchemy_fn else ref_col
```

Reply:
Yeah, I can try this, though I think maybe all the columns also then need to go in the `group by` and things will get funny if the primary value appears more than once in the secondary table. does that sound right?